src/gt4py/next/program_processors/runners/gtfn.py

```diff
@@ -113,103 +114,84 @@ def compilation_hash(otf_closure: stages.ProgramCall) -> int:
     )
 
 
-GTFN_DEFAULT_TRANSLATION_STEP: step_types.TranslationStep[
-    languages.NanobindSrcL, languages.LanguageWithHeaderFilesSettings
-] = gtfn_module.GTFNTranslationStep(
-    enable_itir_transforms=True,
-    use_imperative_backend=False,
-    device_type=core_defs.DeviceType.CPU,
-)
+class GTFNCompileWorkflowFactory(factory.Factory):
+    class Meta:
+        model = recipes.OTFCompileWorkflow
 
-GTFN_GPU_TRANSLATION_STEP: step_types.TranslationStep[
-    languages.NanobindSrcL, languages.LanguageWithHeaderFilesSettings
-] = gtfn_module.GTFNTranslationStep(
-    enable_itir_transforms=True,
-    use_imperative_backend=False,
-    device_type=core_defs.DeviceType.CUDA,
-)
+    class Params:
+        device_type: core_defs.DeviceType = core_defs.DeviceType.CPU
 
-GTFN_DEFAULT_COMPILE_STEP: step_types.CompilationStep = compiler.Compiler(
-    cache_strategy=cache.Strategy.SESSION, builder_factory=compiledb.CompiledbFactory()
-)
+    translation = factory.SubFactory(
+        gtfn_module.GTFNTranslationStepFactory, device_type=factory.SelfAttribute("..device_type")
```

Comment:
What is the meaning of 2 dots `..device_type`?

Reply:
In this context it means "look up the attribute on the parent of the SubFactory": https://factoryboy.readthedocs.io/en/stable/reference.html#parents